### PR TITLE
Create first two modules and perform IC pipeline

### PR DESCRIPTION
**Pull request by jenna-tomkinson**
@gwaybio

Here is the PR for the interstellar project with the download data and ic modules along with performing the first CP pipeline to calculate the IC function per channel.

### CODE REVIEW

**Comment by jenna-tomkinson** [comment]
@gwaybio 

I currently have the first wave of data as gitignored since we will be putting on this data on FigShare in the future and didn't want to break Github with large files.

File: 0.wave1_data/1.cellprofiler_ic_processing/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Create initial LoadData csv to calculate illumination correction function
+
+In this module, we present the process for creation of a LoadData csv and CellProfiler pipeline for calculating an illumination correction function for each channel.
```
Comment (gwaybio) [inline 0.wave1_data/1.cellprofiler_ic_processing/README.md:3]:
```suggestion
In this module, we create a LoadData csv and CellProfiler pipeline for calculating an illumination correction function for each channel.
```

File: 0.wave1_data/1.cellprofiler_ic_processing/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+We are using CellProfiler v4.2.4 for all pipelines.
+
+## Step 2: Install pe2loaddata
+
+Using the instructions from their Github, you can install pe2loaddata by running the code block below in the `0.wave1_data/1.cellprofiler_ic_processing/` directory.
+
+```sh
+git clone https://github.com/broadinstitute/pe2loaddata.git
```
Comment (gwaybio) [inline 0.wave1_data/1.cellprofiler_ic_processing/README.md:25]: We should probably fix the version we're installing. Is there a way to install a specific github hash?

File: 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+name: cp4
+
```
Comment (gwaybio) [inline 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:2]:
```suggestion
```

File: 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+channels:
+- conda-forge
+- anaconda
+- bioconda
+- defaults
+
```
Comment (gwaybio) [inline 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]:
```suggestion
```

Comment (gwaybio) [inline 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]:
```suggestion
```

Comment (gwaybio) [reply 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]: I'm not used to seeing these extra spaces!

Comment (gwaybio) [reply 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]: also, aren't these files usually indented by two spaces?

```yaml
# Like this
dependencies:
  - python=3.8

# Not like this
dependencies:
- python=3.8
```

This is a minor style comment.

**Review by gwaybio — APPROVED** [review]
Looks great 🤩 

My only other comment is to consider naming the loaddata csv something more specific than `wave1.csv`. It would be good to indicate in the file name (in some way) how this file was generated.

File: 0.wave1_data/1.cellprofiler_ic_processing/README.md (hunk 1 above)
Comment (jenna-tomkinson) [reply 0.wave1_data/1.cellprofiler_ic_processing/README.md:3]: :+1:

File: 0.wave1_data/1.cellprofiler_ic_processing/README.md (hunk 2 above)
Comment (jenna-tomkinson) [reply 0.wave1_data/1.cellprofiler_ic_processing/README.md:25]: @gwaybio 

We might be able to install using this I found from here (https://stackoverflow.com/questions/13685920/install-specific-git-commit-with-pip).

```sh
pip install git+https://github.com/broadinstitute/pe2loaddata.git@0426dd6b9d8b3242294a6fbdef7c4e1ee762a4cc
```

File: 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml (hunk 3 above)
Comment (jenna-tomkinson) [reply 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:2]: :+1:

File: 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml (hunk 4 above)
Comment (jenna-tomkinson) [reply 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]: :+1:

Comment (jenna-tomkinson) [reply 0.wave1_data/1.cellprofiler_ic_processing/cellprofiler_environment.yml:8]: I just copied and pasted what was on the CellProfiler wiki for conda install. This is an easy style change I will implement.

**Pull request merged**